readthedocs/builds/tasks.py

```diff
@@ -453,3 +457,150 @@ def send_build_status(build_pk, commit, status, link_to_build=False):
             build.project.slug
         )
         return False
+
+
+@app.task(queue='web')
+def send_build_notifications(version_pk, build_pk, event):
+    version = Version.objects.get_object_or_log(pk=version_pk)
+
+    if not version or version.type == EXTERNAL:
+        return
+
+    build = Build.objects.get(pk=build_pk)
+
+    sender = NotificationSender(
+        version=version,
+        build=build,
+        event=event,
+    )
+    sender.send()
+
+class NotificationSender:
+
+    def __init__(self, version, build, event):
+        self.version = version
+        self.build = build
+        self.project = version.project
+        self.event = event
+
+    def send(self):
+        if self.event == WebHookEvent.BUILD_FAILED:
+            email_addresses = (
+                self.project.emailhook_notifications.all()
+                .values_list('email', flat=True)
+            )
+            for email in email_addresses:
+                try:
+                    self.send_email(email)
+                except Exception:
+                    log.exception(
+                        'Failed to send email notification. email=%s',
+                        email,
+                    )
+
+        old_webhooks = []
+        if self.event in [WebHookEvent.BUILD_FAILED, WebHookEvent.BUILD_PASSED]:
+            old_webhooks = (
+                self.project.webhook_notifications
+                .filter(events__isnull=True)
+            )
+
+        new_webhooks = (
+            self.project.webhook_notifications
+            .filter(events__name=self.event)
+        )
+        for webhook in itertools.chain(old_webhooks, new_webhooks):
+            try:
+                self.send_webhook(webhook)
+            except Exception:
+                log.exception(
+                    'Failed to send webhook. id=%s',
+                    webhook.id,
+                )
+
+    def send_email(self, email):
+        """Send email notifications for build failure."""
+        # We send only what we need from the Django model objects here to avoid
+        # serialization problems in the ``readthedocs.core.tasks.send_email_task``
+        context = {
+            'version': {
+                'verbose_name': self.version.verbose_name,
+            },
+            'project': {
+                'name': self.project.name,
+            },
+            'build': {
+                'pk': self.build.pk,
+                'error': self.build.error,
+            },
+            'build_url': 'https://{}{}'.format(
```

Comment:
I think `https` depends on `settings.PRODUCTION_DOMAIN_USES_HTTPS` or similar name.

Reply:
This was taken from the old notifications code. I think we are fine always using https.

Reply:
there is only `PUBLIC_DOMAIN_USES_HTTPS`, looks like we are using the debug option to decide that protocol to use in other parts of our code.